tiledb/sm/query/readers/dense_reader.cc

```diff
@@ -72,7 +72,9 @@ DenseReader::DenseReader(
     shared_ptr<Logger> logger,
     StrategyParams& params,
     bool remote_query)
-    : ReaderBase(stats, logger->clone("DenseReader", ++logger_id_), params) {
+    : ReaderBase(stats, logger->clone("DenseReader", ++logger_id_), params)
+    , memory_budget_(params.memory_budget().value_or(0))
+    , update_budget_from_config_(!params.memory_budget().has_value()) {
```

Comment:
I would save params.memory_budget() in strategy base as memory_budget_from_query_ and use that instead of the added boolean value.

Reply:
Ok then we can save params.memory_budget() in the dense reader and in the sparse reader class so that we can use memory_budget_from_query_.has_value() instead of the boolean.

Reply:
Done.